src/marshal.c

```diff
@@ -254,29 +350,53 @@ int decode_packet(bot_t *bot, void *_packet_raw, void *_packet_data)
     int32_t packet_size;
     int packet_size_len = varint32(packet_raw, &packet_size);
     packet_raw += packet_size_len;
+    bytes_read += packet_size_len;
 
     uint64_t host_order;
     while(*fmt) {
         size = format_sizeof(*fmt);
         packet_data = (char *)align(packet_data, size);
         switch(*fmt) {
+        case 'k':
+            /* Notch decided it was a good idea not to tell us the length of
+             * the nbt binary data. cNBT expects us the know this length. We
+             * use the number of bytes left in the packet.
```

Comment:
Not sure that using remaining bytes will always work, especially with compression. We'll fix it when it breaks.